dev/Breadcrumb/Breadcrumb.cpp

```diff
@@ -0,0 +1,566 @@
+{
+    const winrt::IDependencyProperty& property = args.Property();
+
+    // TODO: Implement
+    if (property == s_ItemsSourceProperty)
+    {
+        UpdateItemsRepeaterItemsSource();
+    }
+    else if (property == s_ItemTemplateProperty)
+    {
+        UpdateItemTemplate();
+    }
+    else if (property == s_DropdownItemTemplateProperty)
+    {
+        UpdateEllipsisBreadcrumbItemDropdownItemTemplate();
+    }
+}
+
+void Breadcrumb::OnBreadcrumbItemRepeaterLoaded(const winrt::IInspectable&, const winrt::RoutedEventArgs&)
+{
+    if (const auto& breadcrumbItemRepeater = m_itemsRepeater.get())
+    {
+        OnRepeaterCollectionChanged(nullptr, nullptr);
+    }
+}
+
+void Breadcrumb::UpdateItemTemplate()
+{
+    const winrt::IInspectable& newItemTemplate = ItemTemplate();
+    m_breadcrumbElementFactory->UserElementFactory(newItemTemplate);
+}
+
+void Breadcrumb::UpdateEllipsisBreadcrumbItemDropdownItemTemplate()
+{
+    const winrt::IInspectable& newItemTemplate = DropdownItemTemplate();
+
+    // Copy the item template to the ellipsis button too
+    if (const auto& ellipsisBreadcrumbItem = m_ellipsisBreadcrumbItem.get())
+    {
+        if (const auto& itemImpl = winrt::get_self<BreadcrumbItem>(ellipsisBreadcrumbItem))
+        {
+            itemImpl->SetFlyoutDataTemplate(newItemTemplate);
+        }
+    }
+}
+
+void Breadcrumb::UpdateItemsRepeaterItemsSource()
+{
+    m_itemsSourceChanged.revoke();
+    m_itemsSourceChanged2.revoke();
+
+    m_itemsRepeaterItemsSource = winrt::ItemsSourceView(ItemsSource());
+    m_itemsSourceChanged = m_itemsRepeaterItemsSource.CollectionChanged(winrt::auto_revoke, { this, &Breadcrumb::OnRepeaterCollectionChanged });
+
+    if (const auto& breadcrumbItemRepeater = m_itemsRepeater.get())
+    {
+        if (const auto& itemsSource = this->ItemsSource())
+        {
+            const auto& incc = [this, itemsSource]() {
+                return itemsSource.try_as<winrt::INotifyCollectionChanged>();
+            }();
+
+            if (incc)
+            {
+                m_collectionChanged = incc.CollectionChanged({ this, &Breadcrumb::OnRepeaterCollectionChanged });
+                m_notifyCollectionChanged.set(incc);
+            }
+
+            if (const auto& collection = itemsSource.try_as<winrt::IObservableVector<IInspectable>>())
+            {
+                m_itemsSourceChanged2 = collection.VectorChanged(winrt::auto_revoke, { this, &Breadcrumb::OnRepeaterCollectionChanged });
+            }
+        }
+    }
+}
+
+void Breadcrumb::OnRepeaterCollectionChanged(const winrt::IInspectable&, const winrt::IInspectable& args)
+{
+    if (const auto& itemsRepeater = m_itemsRepeater.get())
+    {
+        // A new BreadcrumbIterable must be created as ItemsRepeater compares if the previous
+        // itemsSource is equals to the new one
+        m_itemsIterable = winrt::make_self<BreadcrumbIterable>(ItemsSource());
```

Comment:
Can you add comments about the difference between
m_itemsRepeaterItemsSource = winrt::ItemsSourceView(ItemsSource());
and
m_itemsIterable = winrt::make_self<BreadcrumbIterable>(ItemsSource());
?